fdbserver/storageserver.actor.cpp

```diff
@@ -9616,14 +9616,77 @@ ACTOR Future<Void> update(StorageServer* data, bool* pReceivedUpdate) {
 	}
 }
 
+ACTOR Future<Void> createSstFileForCheckPointMetadata(StorageServer* data,
+                                                      CheckpointMetaData metaData,
+                                                      std::string dir) {
+	state int failureCount = 0;
+	loop {
+		try {
+			ASSERT(metaData.ranges.size() > 0);
+			state std::string localFile = dir + "/metadata_bytes.sst";
+			state std::unique_ptr<IRocksDBSstFileWriter> sstWriter = newRocksDBSstFileWriter();
+			sstWriter->open(localFile);
+			if (sstWriter == nullptr) {
+				return Void();
+			}
+			state std::vector<KeyRange>::iterator iter = metaData.ranges.begin();
+			std::sort(metaData.ranges.begin(), metaData.ranges.end(), [](KeyRange a, KeyRange b) {
+				return a.begin > b.begin;
+			}); // inplacement sorting acceptable?
+			state int64_t numGetRangeQueries = 0;
+			state int64_t numSampledKeys = 0;
+			while (iter != metaData.ranges.end()) {
+				state KeyRange range = *iter;
+				state Key readBegin = range.begin.withPrefix(persistByteSampleKeys.begin);
+				state Key readEnd = range.end.withPrefix(persistByteSampleKeys.begin);
+				loop {
+					try {
+						state RangeResult result = wait(data->storage.readRange(KeyRangeRef(readBegin, readEnd),
+						                                                        SERVER_KNOBS->STORAGE_LIMIT_BYTES,
+						                                                        SERVER_KNOBS->STORAGE_LIMIT_BYTES));
+						numGetRangeQueries++;
+						for (int i = 0; i < result.size(); i++) {
+							numSampledKeys++;
+							sstWriter->write(result[i].key, result[i].value);
+						}
+						if (result.more) {
+							ASSERT(result.readThrough.present());
+							readBegin = keyAfter(result.readThrough.get());
+						} else {
+							break;
+						}
+					} catch (Error& e) {
+						if (failureCount < SERVER_KNOBS->ROCKSDB_CREATE_SST_FILE_RETRY_COUNT_MAX) {
+							failureCount++;
+							continue;
+						} else {
+							throw e;
+						}
+					}
+				}
+				iter++;
+			}
+			sstWriter->finish();
```

Comment:
Can you add `CheckpointMetaData::bytesSampeFile` to record the location of that file.